lib/cext/preprocess.rb

```diff
@@ -49,7 +62,35 @@
                  'websocket_mask.c'    => {:gem   => 'websocket-driver',
                                            :patches => 
                                              [{:match       => /(VALUE .*?)\s+= Qnil;/,
-                                               :replacement => '\1;'}]}}
+                                               :replacement => '\1;'}]},
+                 'http11_parser.c'    => {:gem   => 'puma_http11',
+                                           :patches => 
+                                             [{:match       => /parser->(.*?) = Qnil/,
+                                               :replacement => 'parser->\1 = rb_tr_handle_for_managed(Qnil)'}]},
+                 'puma_http11.c'      => {:gem   => 'puma_http11',
+                                           :patches => 
+                                             [{:match       => /(define.*?,) Qnil }/,
```

Comment:
I would match those more specifically, as they could be valid if used in a function and not just to initialize a global variable.
```c
# define f(N) { (sizeof(N) - 1), N, 0, Qnil }
# define fr(N) { (sizeof(N) - 1), N, 1, Qnil }
```